roiextractors/multisegmentationextractor.py

```diff
@@ -0,0 +1,186 @@
+        return pixel_masks_all
+
+    def get_roi_image_masks(self, roi_ids=None):
+        images_all = []
+        if roi_ids is None:
+            roi_ids = np.array(self._all_roi_ids)
+        else:
+            roi_ids = np.array(roi_ids)
+        seg_id = np.array([self._roi_map[roi_id]['segmentation_id'] for roi_id in roi_ids])
+        roi_id_segmentation = np.array([self._roi_map[roi_id]['roi_id'] for roi_id in roi_ids])
+        for i in np.unique(seg_id):
+            seg_roi_ids = roi_id_segmentation[seg_id == i]
+            images_all.append(self._segmentations[i].get_roi_image_masks(roi_ids=seg_roi_ids))
+        return np.concatenate(images_all, axis=2)
+
+    def get_roi_ids(self):
+        return self._all_roi_ids
+
+    def get_roi_locations(self, roi_ids=None):
+        roi_locations_all = []
+        if roi_ids is None:
+            roi_ids = np.array(self._all_roi_ids)
+        else:
+            roi_ids = np.array(roi_ids)
+        seg_id = np.array([self._roi_map[roi_id]['segmentation_id'] for roi_id in roi_ids])
+        roi_id_segmentation = np.array([self._roi_map[roi_id]['roi_id'] for roi_id in roi_ids])
+        for i in np.unique(seg_id):
+            seg_roi_ids = roi_id_segmentation[seg_id == i]
+            roi_locations_all.append(self._segmentations[i].get_roi_locations(roi_ids=seg_roi_ids))
+        return np.concatenate(roi_locations_all, axis=1)
+
+    def get_num_frames(self):
+        return np.sum([self._segmentations[i].get_num_frames() for i in range(self._no_planes)])
+
+    def get_accepted_list(self):
+        accepted_list_all = []
+        for i in range(self._no_planes):
+            ids_loop = self._segmentations[i].get_accepted_list()
+            accepted_list_all.extend(
+                [j for j in self._all_roi_ids if self._roi_map[j]['roi_id'] in ids_loop])
+        return accepted_list_all
+
+    def get_rejected_list(self):
+        rejected_list_all = []
+        for i in range(self._no_planes):
+            ids_loop = self._segmentations[i].get_rejected_list()
+            rejected_list_all.extend(
+                [j for j in self._all_roi_ids if self._roi_map[j]['roi_id'] in ids_loop])
+        return rejected_list_all
+
+    @staticmethod
+    def write_segmentation(multisegmentation_extractor, savepath, **kwargs):
+        """
+        Write plane wise segmentation
+        Parameters
+        ----------
+        multisegmentation_extractor: MultiSegmentationExtractor
+        savepath: str
+        kwargs: dict
+            example 'to_type': 'Suite2pSegmentationExtractor'
+        """
+        if kwargs.get('to_type'):
+            try:
+                eval('to_type='+kwargs['to_type'])
+            except:
+                raise Exception('enter a valid SegmentationExtractor type')
+        else:
+            to_type = eval(multisegmentation_extractor.segmentations[0].extractor_name)
```

Comment:
I think `to_type` should take a `SegmentationExtractor` class, not a string. In general, try to avoid using `eval`, especially on an unrestricted user input. This opens huge security holes. What if someone put kwargs={'to_type': '[remove force root]'?

Reply:
yes, true! Made changes in the latest commit https://github.com/catalystneuro/roiextractors/pull/36/commits/7de46565e291e704459dba59e0f97a636efdfdf9